pkg/kubernetes/client.go

```diff
@@ -67,6 +68,12 @@ func (g *CallbackClientSet) ClientSet() (Interface, error) {
 	return g.Callback()
 }
 
+// ClientInterface defines the high level client used by eks pkg
+type ClientInterface interface {
+	NewClientSet() (Interface, error)
+	Config() *clientcmdapi.Config
```

Comment:
Do I see correctly that this isn't call by anything anywhere?

Reply:
What is the purpose of this `Config` method?

Reply:
Seems to be used in the cluster creation gitops flow... would love to get rid of this layer of abstraction and just have dedicated NewClientSet / Config methods in the KubeProvider interface instead.

Reply:
Seems to be a similar problem with RawClientInterface which was pre-existing before the refactors in this PR, although that has a lot more methods on it (NewRawResource, ServerVersion, CreateOrReplace)

Reply:
Ah yes. Somehow I missed that usage.